SignalServiceKit/src/Protos/Generated/SSKProto.swift

```diff
@@ -0,0 +1,3163 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+// WARNING: This code is generated. Only edit within the markers.
+
+public enum SSKProtoError: Error {
+    case invalidProtobuf(description: String)
+}
+
+// MARK: - SSKProtoEnvelope
+
+@objc public class SSKProtoEnvelope: NSObject {
+
+	// MARK: - SSKProtoEnvelopeType
+
+	@objc public enum SSKProtoEnvelopeType: Int32 {
+		case unknown = 0
+		case ciphertext = 1
+		case keyExchange = 2
+		case prekeyBundle = 3
+		case receipt = 5
+	}
+
+	private class func SSKProtoEnvelopeTypeWrap(_ value: SignalServiceProtos_Envelope.TypeEnum) -> SSKProtoEnvelopeType {
+		switch value {
+		case .unknown: return .unknown
+		case .ciphertext: return .ciphertext
+		case .keyExchange: return .keyExchange
+		case .prekeyBundle: return .prekeyBundle
+		case .receipt: return .receipt
+		}
+	}
+
+	private class func SSKProtoEnvelopeTypeUnwrap(_ value: SSKProtoEnvelopeType) -> SignalServiceProtos_Envelope.TypeEnum {
+		switch value {
+		case .unknown: return .unknown
+		case .ciphertext: return .ciphertext
+		case .keyExchange: return .keyExchange
+		case .prekeyBundle: return .prekeyBundle
+		case .receipt: return .receipt
+		}
+	}
+
+	@objc public let type: SSKProtoEnvelopeType
+	@objc public let source: String?
+	@objc public let sourceDevice: UInt32
+	@objc public let relay: String?
+	@objc public let timestamp: UInt64
+	@objc public let legacyMessage: Data?
+	@objc public let content: Data?
```

Comment:
One of the limitations is that our proto schemas don't use the `required` "rule" to indicate which fields are required.  One option would be to use schema comments to decorate required fields, e.g.:

```
// @required
optional Type   type          = 1;
// @required
optional string source        = 2;
// @required
optional uint32 sourceDevice  = 7;
optional string relay         = 3;
```

But I'm inclined to leave things as they stand. I suspect that the "optionality" of class properties won't require much change if any in our parsing code.